Source/Core/Common/GL/GLInterface/BGL.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/GL/GLInterface/BGL.h"
+
+#include <GLView.h>
+#include <Size.h>
+#include <Window.h>
+
+BGLView* GLContextBGL::m_current = nullptr;
+
+GLContextBGL::~GLContextBGL()
+{
+  if (!m_window)
+    delete m_gl;
+}
+
+bool GLContextBGL::Initialize(const WindowSystemInfo& wsi, bool stereo, bool core)
+{
+  m_window = static_cast<BWindow*>(wsi.render_window);
+
+  m_gl = new BGLView(m_window ? m_window->Bounds() : BRect(), "GLContextBGL", B_FOLLOW_ALL_SIDES, 0,
+    BGL_RGB | BGL_DOUBLE | BGL_ALPHA);
+  if (m_window)
+    m_window->AddChild(m_gl);
+
+  m_opengl_mode = Mode::OpenGL;
+
+  m_gl->LockLooper();
+  BRect size = m_gl->Frame();
+  m_gl->UnlockLooper();
+
+  m_backbuffer_width = size.IntegerWidth();
+  m_backbuffer_height = size.IntegerHeight();
+
+  MakeCurrent();
+
+  return true;
+}
+
+bool GLContextBGL::IsHeadless() const
+{
+  return !m_window;
+}
+
+bool GLContextBGL::MakeCurrent()
+{
+  if (m_current)
+    m_current->UnlockGL();
+  m_gl->LockGL();
+  m_current = m_gl;
+  return true;
+}
+
+bool GLContextBGL::ClearCurrent()
+{
+  if (!m_current)
+    return true;
+
+  //assert(m_gl == m_current);
```

Comment:
This should either be uncommented or removed entirely.

Reply:
I commented it because I was not sure what assert facilities you generally use, i.e. if I should use something besides `<cassert>` (I did not immediately see anything.) Should I just include the header and uncomment it, then?

Reply:
`#include "Common/Assert.h"` and ASSERT(). (`ASSERT(m_gl == m_current);` for example)